src/ape_cache/models.py

```diff
@@ -0,0 +1,82 @@
+from sqlalchemy import JSON, LargeBinary, BigInteger, Column, ForeignKey, Integer, Numeric
+
+from sqlalchemy.types import String, TypeDecorator
+
+from ape.api.address import BaseAddress, Address
+from .base import Base
+
+
+class HexByteString(TypeDecorator):
+    """Convert Python bytestring to string with hexadecimal digits and back for storage."""
+
+    impl = String
+
+    def process_bind_param(self, value, dialect):
+        if not isinstance(value, bytes):
+            raise TypeError("HexByteString columns support only bytes values.")
+        return value.hex()
+
+    def process_result_value(self, value, dialect):
+        return bytes.fromhex(value) if value else None
+
+
+class AddressString(TypeDecorator):
+    """
+    Convert Python :class:`~ape.types.AddressType` string to
```

Comment:
This says `AddressType` here but the logic is using `BaseAddress`.

Reply:
Good catch, we originally had the logic backwards, missed the docstring

Reply:
[45fcdfc](https://github.com/ApeWorX/ape/pull/680/commits/45fcdfcba6ed6a53450b7839ccc8785a2c6dfd97)